src/Http/Http.Extensions/gen/StaticRouteHandlerModel/EndpointParameter.cs

```diff
@@ -389,6 +451,74 @@ private static string GetEscapedParameterName(AttributeData attribute, string pa
         }
     }
 
+    private static bool TryGetAsParametersConstructor(INamedTypeSymbol type, out bool? isParameterlessConstructor, [NotNullWhen(true)] out IEnumerable<ConstructorParameter>? matchedParameters)
+    {
+        isParameterlessConstructor = null;
+        matchedParameters = null;
+        if (type.IsAbstract)
+        {
+            return false;
+        }
+
+        var constructors = type.Constructors.Where(constructor => constructor.DeclaredAccessibility == Accessibility.Public && !constructor.IsStatic);
+        var numOfConstructors = constructors.Count();
+        // When leveraging parameterless constructors, we want to ensure we only emit for writable
+        // properties. We do not have this constraint if we are leveraging a parameterized constructor.
+        var properties = type.GetMembers().OfType<IPropertySymbol>().Where(property => property.DeclaredAccessibility == Accessibility.Public);
+        var writableProperties = properties.Where(property => !property.IsReadOnly);
```

Comment:
What does `IsReadOnly` return for `public int P { get; private set; }`?

Reply:
`IsReadOnly` is true for public properties with a private setter. I'll add a case to `RequestDelegatePopulatesFromParameterListAndSkipReadOnlyProperties` to cover this.

Reply:
This end up being a great inquiry because I discovered a bug in the RDF implementation. RDF only checks for `PropertyInfo.CanWrite` which only checks to see if a `set` accessor is defined, not if it is accessible. This meant we were generating invalid code for scenarios where users used private setters. I resolved this in the reflection-based implementation by checking that `GetSetMethod(nonPublic: true)` returned a non-null value.

Reply:
Neat!